pkg/collector/reconcile/configmap.go

```diff
@@ -103,7 +103,7 @@ func desiredTAConfigMap(params Params) (corev1.ConfigMap, error) {
 		labels["app.kubernetes.io/version"] = "latest"
 	}
 
-	prometheusReceiverConfig, err := ta.ConfigToPromConfig(params.Instance.Spec.Config)
+	prometheusReceiverConfig, err := ta.UnescapeDollarSignsInPromConfig(params.Instance.Spec.Config)
```

Comment:
I think it's worth adding a comment here as well. Without context it's not clear at all why we'd need to unescape these here.

Reply:
Done. Added a comment with the reason to unescape.